account_vat_period_end_statement/models/account.py

```diff
@@ -508,12 +508,18 @@ def compute_amounts(self):
                 [('date', '<', statement.date)], order='date desc')
             if prev_statements:
                 prev_statement = prev_statements[0]
-                if (
-                    prev_statement.residual > 0 and
-                    prev_statement.authority_vat_amount > 0
-                ):
+                if prev_statement.authority_vat_amount > 0:
+                    prev_debit_vat_amount = prev_statement.authority_vat_amount
+                    authority_vat_account = statement.authority_vat_account_id
+                    move_lines = self.env['account.move.line'].search(
+                        [('account_id', '=', authority_vat_account.id),
+                         ('date', '>', prev_statement.date),
+                         ('date', '<=', statement.date)]
+                    )
+                    for line in move_lines:
+                        prev_debit_vat_amount -= line.amount_residual
```

Comment:
Nel comportamento desiderato hai scritto:
> Il campo `previous_debit_vat_amount` riporta il campo `authority_vat_amount` del periodo precedente meno i movimenti di pagamento effettuati nello stesso (in modo da arrivare allo 0,00).

Qui invece mi sembra stia succedendo qualcosa di diverso: dal campo `authority_vat_amount` della dichiarazione precedente sottrai tutti i movimenti che hanno data compresa tra le due dichiarazioni, non quelli 'effettuati nello stesso' e non mi sembra ci sia un controllo se sono pagamenti.
Come mai questa differenza?

Perché poi si dovrebbe arrivare allo 0,00?
Nell'esempio in descrizione hai scritto che potrebbe portare l'`amount` da 4000 a 3000, quindi non è detto che sia sempre 0.

Puoi aggiungere un test per chiarire qual'è il problema che le tue modifiche risolvono?

Reply:
Partiamo da qui:
> Perché poi si dovrebbe arrivare allo 0,00?
> Nell'esempio in descrizione hai scritto che potrebbe portare l'amount da 4000 a 3000, quindi non è detto che sia sempre 0.
Nell'esempio che ho fatto, `amount` passa da 4000 a 3000 perché il campo `previous_credit_vat_amount` della liquidazione precedente è valorizzato a 1000, non c'entra `previous_debit_vat_amount` che sarà 0.
Così com'è adesso, il debito della liquidazione nel periodo precedente viene riportato senza tenere conto che, durante il periodo attuale, è stato eseguito uno o più pagamenti a saldo di questo.

> non mi sembra ci sia un controllo se sono pagamenti.
Come mi ha riferito il funzionale, sul conto Erario IVA i movimenti sono quelli di IVA credito, IVA debito, saldo a credito/debito e pagamenti.
Se non erro, ma mi informerò per sicurezza, nel periodo tra la liquidazione precedente e quella che si sta creando, su questo conto ci possono essere solo movimenti di saldo del debito verso l'Erario.

> Puoi aggiungere un test per chiarire qual'è il problema che le tue modifiche risolvono?
Certo, aiuterà a capire meglio il caso d'uso